forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -12,31 +12,78 @@
 from forest.benchmarking.utils import prepare_prod_pauli_eigenstate
 from pyquil import Program
 from pyquil.api import BenchmarkConnection, QuantumComputer
-from pyquil.operator_estimation import ExperimentSetting, TomographyExperiment, \
-    TensorProductState, plusX, minusX, plusY, minusY, plusZ, minusZ
+from pyquil.operator_estimation import ExperimentResult, ExperimentSetting, TomographyExperiment, \
+    TensorProductState, measure_observables, plusX, minusX, plusY, minusY, plusZ, minusZ
 from pyquil.paulis import PauliTerm, PauliSum, sI, sX, sY, sZ
 
 
-def calibrate_readout_imperfections(pauli: PauliTerm, quantum_machine: QuantumComputer,
-                                    var: float = 0.01):
-    """
-    Compute the expectation value and variance of measuring a pauli without any pre or post circuit.
-    This can be used to attribute lack of of visibility to readout imperfections.
-
-    TODO:
-      + look into whether we can calibrate multiple PauliTerm with one call
-      + test how well this calibration works with correlated readout errors
-
-    :param pauli: The Pauli expectation to calibrate.
-    :param quantum_machine: The QVM or QPU to use for calibration.
-    :param var: The desired variance of the result.
-    :return: The expectation, variance, and number of shots of the expectation value of a +1
-     eigenstate of pauli.
-    """
-    this_prog = prepare_prod_pauli_eigenstate(pauli)
-    expectation, variance, count = est.estimate_locally_commuting_operator(
-        this_prog, PauliSum([pauli]), var, quantum_machine)
-    return expectation[0], variance[0, 0].real, count
+@dataclass 
+class DFEexperiment:
+    """Experiments to obtain an estimate of fidelity via DFE"""
+
+    exp: TomographyExperiment
+    """The experiment to be performed"""
+
+    type: str
+    """Type of fidelity to be estimate (state or process)"""
+
+
+@dataclass
+class DFEdata:
+    """Experimental data from a DFE experiment"""
+
+    res: List[ExperimentResult]
+    """The experimental results"""
+
+    in_states: List[str]
+    """The input tensor product states being acted on by the `program`"""
+
+    program: Program
+    """The pyquil Program to perform DFE on"""
+
+    out_pauli: List[str]
+    """The expected output Pauli operators after the program acts on the corresponding `in_pauli`"""
+
+    pauli_point_est: List[float]
+    """Point estimate of Pauli operators"""
+
+    pauli_var_est: List[float]
```

Comment:
An ExperimentResult stores stddev.

Reply:
For estimators I think we want to report standard errors, not standard deviations.  If `ExperimentResult` stores std deviation instead of std error, then we need to fix `ExperimentResult`.

Reply:
`ExperimentResult` is a generic dataclass that will store anything in the `std_dev` field you feed into it. pyQuil's `measure_observables` will feed the standard error, computed as Sqrt[var(data) / num_shots], into that field.

Reply:
There are a bunch of tests in the pyQuil version that will test for the correct value of the standard error, with or without symmetrization/calibration. See e.g. https://github.com/rigetti/pyquil/blob/master/pyquil/tests/test_operator_estimation.py#L1413

Reply:
So `ExperimentResults` computes std errors, but calls them std deviations -- IMHO it computes the right thing, and reports it under the wrong name. @joshcombes it sounds to me like we should have a pyQuil PR that changes the `ExperimentResults` fields to be called `stderr`. Thoughts?

Reply:
Personally, I think that makes sense. Btw, all of this discussion also touches upon a larger conversation to be had: whether `TomographyExperiment` is the correct name/data-structure that should exist at the level of pyQuil, whose entire machinery around `measure_observables` is built not just to be applicable to tomography experiments, but also calculating expectation values of general observables.

Reply:
> So `ExperimentResults` computes std errors, but calls them std deviations -- IMHO it computes the right thing, and reports it under the wrong name. @joshcombes it sounds to me like we should have a pyQuil PR that changes the `ExperimentResults` fields to be called `stderr`. Thoughts?

Totally agree!

Reply:
> Personally, I think that makes sense. Btw, all of this discussion also touches upon a larger conversation to be had: whether `TomographyExperiment` is the correct name/data-structure that should exist at the level of pyQuil, whose entire machinery around `measure_observables` is built not just to be applicable to tomography experiments, but also calculating expectation values of general observables.

Yes. I'll make a pyQuil issues.

Reply:
rigetti/pyquil#864